Shared/ViewModels/ActiveSessionsViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Combine
+import Foundation
+import JellyfinAPI
+
+class ActiveSessionsViewModel: ViewModel {
+    @Published
+    var sessions: [SessionInfo] = []
+    @Published
+    var isLoading = false
+    @Published
+    var error: Error?
+
+    private var timerCancellable: AnyCancellable?
+
+    override init() {
+        super.init()
+        startTimer()
+    }
+
+    func startTimer() {
+        timerCancellable = Timer.publish(every: 2.0, on: .main, in: .default)
+            .autoconnect()
+            .sink { [weak self] _ in
+                self?.loadSessions()
+            }
+    }
+
+    func loadSessions() {
+        isLoading = true
+        error = nil
+
+        Task {
+            do {
+                let fetchedSessions = try await get()
+                DispatchQueue.main.async {
```

Comment:
As a tip when inside an `async` context, use the following instead to switch to the main actor:

```swift
await MainActor.run {
// stuff on main thread
}
```